mflowgen/test/test_module.sh

```diff
@@ -323,27 +359,24 @@ if [ "$copy_list" ]; then
     # Copy desired info from gold cache
     for step in ${copy_list[@]}; do
         
-        # Ugh stupid special case TODO/FIXME need a '--use_stash' arg now I guess
-        if [ "$step" == "rtl" ]; then
-            if [ "$final_module" == "tile_array" ]; then
-                # alex hack for fixing/preventing bad rtl
-                set -x
-                mflowgen stash link --path /home/ajcars/tile-array-rtl-stash/2020-0724-mflowgen-stash-75007d
-                mflowgen stash pull --hash 2fbc7a
-                set +x
-                continue
-            fi
-        fi
-
         # Expand aliases e.g. "syn" -> "synopsys-dc-synthesis"
         # echo "  $step -> `step_alias $step`"
         step=`step_alias $step`
     
-        cache=`cd $gold/*${step}; pwd` || FAIL=true
+        # NOTE if cd command fails, pwd (disastrously) defaults to current dir
+        # cache=`cd $gold/*${step}; pwd` || FAIL=true
+        # if [ "$FAIL" == "true" ]; then
+        #     echo "***ERROR could not find cache dir '$gold'"; exit 13
+        # fi
+
+        cache=`cd $gold/*${step}` || FAIL=true
         if [ "$FAIL" == "true" ]; then
-            echo "***ERROR could not find cache dir '$gold'"; exit 13
+            echo "WARNING Could not find cache for step '${step}'"
+            echo "Will try and go on without it..."
+            continue
         fi
-        
+
+        cache=`cd $gold/*${step}; pwd`
         echo "    cp -rpf $cache ."
         cp -rpf $cache .
```

Comment:
Seems like we should really be using mflowgen stashes rather than manually managing these caches. Is there a reason you're not?

Reply:
Using mflowgen stashes will make it easier for us to realize our goal of developers being able to use stashed gold versions of all intermediate steps (generated by buildkite) for quicker PD iterations.

Reply:
> Seems like we should really be using mflowgen stashes rather than manually managing these caches. Is there a reason you're not?

Agreed, we should probably use stashes, that's what they're for, right? However: I understand "cp" and it works, and I'm not sure I understand stash completely (yet). Maybe after this pull is finalized, someone can help me figure it out...